typings/glamorous.d.ts

```diff
@@ -40,6 +40,12 @@ interface ThemeProps {
 
 export class ThemeProvider extends React.Component<ThemeProps, any> { }
 
+export declare function withTheme
+  <P>(
+    component:
+      | React.ComponentClass<P>
```

Comment:
Ahh I see. you have to remove the function call. Instead of `withTheme(ClassToWrap)({})`, it should be `withTheme(ClassToWrap)`.

I gotta admit this is not the best way to write test cases to validate the Typescript definitions and it's very messy. If you stumble upon anything better, please do let us know!